sound/soc/codecs/hdac_hdmi.c

```diff
@@ -1867,8 +1867,8 @@ static int hdmi_codec_prepare(struct device *dev)
 	 * is received. So setting power state is ensured without using loop
 	 * to read the state.
 	 */
-	snd_hdac_codec_read(hdev, hdev->afg, 0,	AC_VERB_SET_POWER_STATE,
-							AC_PWRST_D3);
+	snd_hdac_codec_write(hdev, hdev->afg, 0, AC_VERB_SET_POWER_STATE,
```

Comment:
@bardliao the comment just above this line contradicts exactly what you're doing here. This commit explains why codec_read should be instead of codec_write() 1b377ccddd14cd04df4b9523a426b34f928002bc